src/test/java/it/pagopa/transactions/services/v1/CircuitBreakerTest.java

```diff
@@ -54,9 +64,315 @@ class CircuitBreakerTest {
+            new NodoErrorException(new CtFaultBean()),
+            "it.pagopa.transactions.exceptions.InvalidNodoResponseException",
+            new InvalidNodoResponseException("")
+    );
+
+    static {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        try {
+            resilience4jConfiguration = mapper.readTree(new File("./src/main/resources/application.yml"));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionsForRetry(String retryInstanceName) {
+        return StreamSupport.stream(
+                Spliterators.spliteratorUnknownSize(
+                        resilience4jConfiguration
+                                .get("resilience4j.retry")
+                                .get("instances")
+                                .get(retryInstanceName)
+                                .get("ignoreExceptions")
+                                .elements(),
+                        Spliterator.ORDERED
+                ),
+                false
+        )
+                .map(json -> {
+                    String exceptionName = json.asText();
+                    return Arguments.of(
+                            Optional
+                                    .ofNullable(exceptionMapper.get(exceptionName))
+                                    .orElseThrow(
+                                            () -> new RuntimeException(
+                                                    "No mapping found for exception: %s".formatted(exceptionName)
+                                            )
+                                    ),
+                            retryInstanceName
+                    );
+                }
+
+                );
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForNewTransactionRetry() {
+        return getIgnoredExceptionsForRetry("newTransaction");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForGetTransactionInfoRetry() {
+        return getIgnoredExceptionsForRetry("getTransactionInfo");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForRequestTransactionAuthorizationRetry() {
+        return getIgnoredExceptionsForRetry("requestTransactionAuthorization");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForUpdateTransactionAuthorizationRetry() {
+        return getIgnoredExceptionsForRetry("updateTransactionAuthorization");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForCancelTransactionRetry() {
+        return getIgnoredExceptionsForRetry("cancelTransaction");
+    }
+
+    private static Stream<Arguments> getIgnoredExceptionForAddUserReceiptRetry() {
+        return getIgnoredExceptionsForRetry("addUserReceipt");
+    }
+
+    @ParameterizedTest
+    @MethodSource("getIgnoredExceptionForNewTransactionRetry")
+    @Order(0)
```

Comment:
Why is an explicit order needed? Is it to control the circuit breaker behaviour?

Reply:
Yes, I want to run first all tests that will not open circuit breaker (ignored exceptions) then the last one that tests circuit breaker opening :)